proxy/tcp/sni_proxy_test.go

```diff
@@ -0,0 +1,58 @@
+package tcp
+
+import (
+	"testing"
+
+	"github.com/fabiolb/fabio/assert"
+)
+
+func TestCreateClientHelloBufferNotTLS(t *testing.T) {
+	assertEqual := assert.Equal(t)
+
+	testCases := [][]byte{
+		// not enough data
+		{0x16, 0x03, 0x01, 0x00, 0x00, 0x01, 0x00, 0x05},
+
+		// not tls record
+		{0x15, 0x03, 0x01, 0x01, 0xF4, 0x01, 0x00, 0x01, 0xeb},
+
+		// too large record
+		//                |---------|
+		{0x16, 0x03, 0x01, 0x40, 0x01, 0x01, 0x00, 0x01, 0xec},
+
+		// zero record length
+		//                |----------|
+		{0x16, 0x03, 0x01, 0x00, 0x00, 0x01, 0x00, 0x01, 0xec},
+
+		// not client hello
+		//                            |----|
+		{0x16, 0x03, 0x01, 0x01, 0xF4, 0x02, 0x00, 0x01, 0xeb},
+
+		// bad handshake length
+		//                                  |----- 0 --------|
+		{0x16, 0x03, 0x01, 0x00, 0xaa, 0x01, 0x00, 0x00, 0x00},
+
+		// Fragmentation (handshake larger than record)
+		//                |-  500 ---|      |----- 497 ------|
+		{0x16, 0x03, 0x01, 0x01, 0xF4, 0x01, 0x00, 0x01, 0xf1},
+	}
+
+	for i := 0; i < len(testCases); i++ {
```

Comment:
I really like these test cases. Can you move them to the `tls_clienthello_test.go` since they test the parsing.

When you refactor the method to just return the size of the buffer you can test for that as well. My suggestion is:

```go
tests := []struct {
    name string
    data []byte
    fail bool
    len int
}{
    {
        name: "bad handshake length",
        data: ...,
        fail: true,
    },
    ...
}

for _, tt := range tests {
    t.Run(tt.name, func(t *testing.T) {
        n, err := clientHelloBufferSize(tt.data)
        if tt.fail && err == nil {
            t.Fatal("should fail")
        }
        if got, want := n, tt.len; !tt.fail && got != want {
            t.Fatalf("got buffer size %d want %d", got, want)
        }
    }
}
```

Also I